src/einsteinpy/geodesic/utils.py

```diff
@@ -1,261 +1,223 @@
-    ph : float
-        phi-component of 3-Position
+        r-component of 4-Position
+    theta : float
+        theta-component of 4-Position
+    a : float
+        Spin Parameter
+        :math:`0 \\le a \\le 1`
 
     Returns
     -------
-    tuple
-        3-Tuple, containing the coordinates in cartesian form
+    float
+        The value of :math:`r^2 + a^2 * \\cos^2(\\theta)`
 
     """
-    xs = r * np.sin(th) * np.cos(ph)
-    ys = r * np.sin(th) * np.sin(ph)
-    zs = r * np.cos(th)
+    sigma = (r ** 2) + ((a * np.cos(theta)) ** 2)
 
-    return xs, ys, zs
+    return sigma
 
 
-def _f_vec(ld, y, params):
+def delta(r, a, Q=0):
     """
-    Evaluates expressions for the RHS of the dynamic equations,
-    from the Kerr Hamiltonian, to be used with ``_verlet_step()``
-
-    Source: `Fuerst and Wu, 2004: <hhttps://www.aanda.org/articles/aa/pdf/2004/36/aa0814.pdf>`_
+    Returns the value of :math:`r^2 - r_s r + a^2 + r_Q^2`
+    Uses natural units, with :math:`c = G = M = k_e = 1`
 
     Parameters
     ----------
-    ld : float
-        Affine Parameter, Lambda (Step-size)
-    y : numpy.ndarray
-        Length-6 array, containing the initial position and momentum of the test particle
-    params : array_like
-        Length-3 Array, containing - Black Hole Spin Parameter, `a`, Test Particle Energy, `E` and
-        Test Particle Rest Mass, `mu`
+    r : float
+        r-component of 4-Position
+    a : float
+        Spin Parameter
+        :math:`0 \\le a \\le 1`
+    Q : float
+        Charge on gravitating body
+        Defaults to ``0``
 
     Returns
     -------
-    yn : numpy.ndarray
-        Length-6 array, containing the RHS of the dynamic equations
+    float
+        The value of :math:`r^2 - r_s r + a^2 + r_Q^2`
 
     """
-    yn = np.zeros(shape=y.shape)
+    delta = (r ** 2) - (2 * r) + (a ** 2) + Q ** 2
 
-    r, th, ph = y[:3]
-    pr, pth, pph = y[3:]
-    a, E, mu = params
+    return delta
 
-    r2 = r ** 2
-    pr2 = pr ** 2
-    pth2 = pth ** 2
-    pph2 = pph ** 2
-    a2 = a ** 2
-    E2 = E ** 2
-    sint = np.sin(th)
-    sint2 = sint ** 2
-    cost = np.cos(th)
-    sg = r2 + a2 * (cost ** 2)
-    dl = r2 - 2 * r + a2
-    kappa = pth2 + pph2 * (1 / sint2) + a2 - (E2 * sint2 + mu)
-
-    # Eqs. 21, 22, 27, 30, 31 in Source
-    yn[0] = (dl * pr) / sg
-    yn[1] = pth / sg
-    yn[2] = (a * (-a * pph + 2 * E * r) + pph * (1 / sint2) * dl) / (dl * sg)
-    yn[3] = (1 / (sg * dl)) * (
-        ((r2 + a2) * mu - kappa) * (r - 1)
-        + r * dl * mu
-        + 2 * r * (r2 + a2) * E2
-        - 2 * a * E * pph
-    ) - (2 * pr2 * (r - 1)) / sg
-    yn[4] = ((sint * cost) / sg) * ((pph2 / (sint ** 4)) - a2 * (E2 + mu))
-    yn[5] = 0.0
-
-    return yn
-
-
-def _verlet_step(ld, y, params):
-    """
-    Synchornized Symplectic VerletLeapfrog Integrator
-    Advances integration by one step
 
-    Source: `Wikipedia: <https://en.wikipedia.org/wiki/Leapfrog_integration#Algorithm>`_
+def _sch(x_vec, *params):
```

Comment:
I am not very happy with this for now.

Reply:
We need metric integration

Reply:
Certainly. Right now, the only hindrance is the unit system. If we standardize to natural units for the numerical part of EPy, many of these hacky workarounds would vanish. Things like `metric` being `str` or `metric_params` or `_P()` that computes 4-Velocity, being in `geodesic`, instead of `coordinates`, would either be removed or be in their proper places.

Reply:
In natural units, we can even decouple coordinate systems from the metric, in the rudimentary sense. I have realized, we do not really need the metrics defined in all coordinate systems or the conversions between them. I mean, in some cases, we cannot convert between co-ordinate systems, without the knowledge of the metric tensor in at least one of them.

For getting geodesics on a manifold, working in the system, that is computationally most efficient, is all we need. In case of vacuum solutions, it's mostly Boyer-Lindquist or a variation of it. For perturbed spacetimes, it's either a modification of BL or Kerr-Schild coordinates. Any locally flat system, like Cartesian or Spherical or similar systems, where conversions are straightforward are also needed to "graph" the results as it would appear to a distant observer.

In any case, if a particular spacetime needs a particular coordinate system, we can add it as needed. Conversions need not be defined, as not all conversions make sense without a physical perspective.

Reply:
We are proceeding with this ^.